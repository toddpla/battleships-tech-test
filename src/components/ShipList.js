import React from 'react'
import ShipListItem from './ShipListItem'

export class ShipList extends React.Component {

  render() {
    return (
      <div className="ship-list">
        <h1>Ships</h1>
        <article>
          {this.props.ships.map((ship, i) => (
            <ShipListItem
              key={i}
              index={i}
              selectedClass={i == this.props.selectedShip ? "ship-selected" : "ship-unselected"}
              handleSelect={this.props.handleSelect}
              {...ship}
              />
          ))}
        </article>
      </div>
    )
  }
}

export default ShipList
