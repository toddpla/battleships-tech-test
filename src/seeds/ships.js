export default [{
  id: 1,
  name: 'Carrier',
  size: 5
}, {
  id: 2,
  name: 'Battleship',
  size: 4
}, {
  id: 3,
  name: 'Cruiser',
  size: 3
}, {
  id: 4,
  name: 'Submarine',
  size: 3
}, {
  id: 5,
  name: 'Destroyer',
  size: 2
}]