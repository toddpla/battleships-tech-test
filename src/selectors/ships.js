export const unSelectedShips = ships => ships.filter(ship => ship.squares.length === 0)
