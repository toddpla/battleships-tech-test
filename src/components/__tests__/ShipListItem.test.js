import React from 'react'
import { ShipList } from '../ShipList'
import renderer from 'react-test-renderer';
import ships from '../../seeds/ships'

test('renders correctly', () => {
  const shipList = renderer.create(<ShipList ships={ships}/>).toJSON();
  expect(shipList).toMatchSnapshot();
});
