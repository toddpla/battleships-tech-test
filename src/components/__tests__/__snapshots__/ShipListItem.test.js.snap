// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="ship-list"
>
  <h1>
    Ships
  </h1>
  <article>
    <div
      className="ship ship-unselected"
      id="ship-0"
      onClick={[Function]}
    >
      Carrier
    </div>
    <div
      className="ship ship-unselected"
      id="ship-1"
      onClick={[Function]}
    >
      Battleship
    </div>
    <div
      className="ship ship-unselected"
      id="ship-2"
      onClick={[Function]}
    >
      Cruiser
    </div>
    <div
      className="ship ship-unselected"
      id="ship-3"
      onClick={[Function]}
    >
      Submarine
    </div>
    <div
      className="ship ship-unselected"
      id="ship-4"
      onClick={[Function]}
    >
      Destroyer
    </div>
  </article>
</div>
`;
